Cross-compiling fails for 32-bit mpv.com during link phase (multilib)

### mpv version and platform

Building from git HEAD, with GCC 8.1.0 and MinGW-w64 5.x (git) in a multilib setup on Ubuntu 18.04.

### Reproduction steps

```
git clone git://github.com/mpv-player/mpv.git && \
cd mpv && \
./bootstrap.py && \
    CFLAGS="-DHAVE_GPL -m32 -mfpmath=sse -march=pentium3 -msse -mtune=pentium3 \
    -DCACA_STATIC -DVS_CORE_EXPORTS" LDFLAGS="-m32 -L/usr/i686-w64-mingw32/lib" \
    WINDRES="x86_64-w64-mingw32-windres -F pe-i386" \
    PKG_CONFIG_LIBDIR="$HOME/ffmpeg_build/32bit/lib:/usr/i686-w64-mingw32/lib" \
    PKG_CONFIG_PATH="$HOME/ffmpeg_build/32bit/lib/pkgconfig:/usr/i686-w64-mingw32/lib/pkgconfig" \
    TARGET=x86_64-w64-mingw32 ./waf configure --prefix=$HOME/mpv_build/32bit \
    --disable-debug-build --enable-static-build --enable-openal --enable-sdl2 \
    --enable-dvdread --enable-dvdnav --enable-cdda --disable-manpage-build \
    --disable-pdf-build && \
./waf build && \
./waf install && \
strip ~/mpv_build/32bit/bin/mpv.exe && \
strip ~/mpv_build/32bit/bin/mpv.com
```

### Expected behavior

It builds successfully.

### Actual behavior

```
[224/227] Compiling video/out/vo_lavc.c
[225/227] Compiling filters/f_output_chain.c
[226/227] Linking build/mpv.com
/usr/bin/../lib/gcc/x86_64-w64-mingw32/8.1.0/../../../../x86_64-w64-mingw32/bin/ld: i386:x86-64 architecture of input file `osdep/win32-console-wrapper.c.25.o' is incompatible with i386 output
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x15): undefined reference to `_imp_GetLastError'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x4d): undefined reference to `_imp_FormatMessageW'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x5a): undefined reference to `_imp___iob_func'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x76): undefined reference to `fwprintf'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x84): undefined reference to `_imp_LocalFree'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0xca): undefined reference to `memset'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0xdd): undefined reference to `_imp_GetStdHandle'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0xf2): undefined reference to `_imp_GetStdHandle'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x107): undefined reference to `_imp_GetStdHandle'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x133): undefined reference to `_imp_GetStartupInfoW'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x162): undefined reference to `memset'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x1ba): undefined reference to `_imp_CreateProcessW'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x1e1): undefined reference to `_imp_WaitForSingleObject'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x1f5): undefined reference to `_imp_GetExitCodeProcess'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x205): undefined reference to `_imp_CloseHandle'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x215): undefined reference to `_imp_CloseHandle'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x24e): undefined reference to `_imp_GetCommandLineW'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x270): undefined reference to `_imp_GetModuleFileNameW'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x283): undefined reference to `wcsrchr'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x296): undefined reference to `wcscpy'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x2ab): undefined reference to `_imp_SetEnvironmentVariableW'
/usr/i686-w64-mingw32/lib/libmingw32.a(lib32_libmingw32_a-crt0_w.o): In function `wmain':
/home/qyot27/mingw-packages/mingw-w64/mingw-w64-crt/crt/crt0_w.c:23: undefined reference to `wWinMain@16'
collect2: error: ld returned 1 exit status

Waf: Leaving directory `/home/qyot27/mpv-build-deps/mpv/build'
Build failed
 -> task in 'mpv' failed with exit status 1 (run with -v to display more information)
```
The output of *./waf build -v*
```
[mpv:$] ./waf build -v
Waf: Entering directory `/home/qyot27/mpv-build-deps/mpv/build'
[1/1] Compiling version.sh
19:35:51 runner ' sh ../version.sh --cwd="/home/qyot27/mpv-build-deps/mpv" --versionh="version.h" '
[211/227] Compiling osdep/mpv.rc
19:35:53 runner ['x86_64-w64-mingw32-windres', '-F', 'pe-i386', '--include-dir=/home/qyot27/mpv-build-deps/mpv/build', '--include-dir=/home/qyot27/mpv-build-deps/mpv', '--codepage=65001', '../osdep/mpv.rc', 'osdep/mpv-rc.o']
[226/227] Linking build/mpv.com
19:35:53 runner ['/usr/bin/x86_64-w64-mingw32-gcc', '-Wl,--enable-auto-import', 'osdep/win32-console-wrapper.c.25.o', '-o/home/qyot27/mpv-build-deps/mpv/build/mpv.com', '-Wl,-Bstatic', '-Wl,-Bdynamic', '-m32', '-L/usr/i686-w64-mingw32/lib', '-municode', '-mconsole']
/usr/bin/../lib/gcc/x86_64-w64-mingw32/8.1.0/../../../../x86_64-w64-mingw32/bin/ld: i386:x86-64 architecture of input file `osdep/win32-console-wrapper.c.25.o' is incompatible with i386 output
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x15): undefined reference to `_imp_GetLastError'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x4d): undefined reference to `_imp_FormatMessageW'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x5a): undefined reference to `_imp___iob_func'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x76): undefined reference to `fwprintf'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x84): undefined reference to `_imp_LocalFree'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0xca): undefined reference to `memset'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0xdd): undefined reference to `_imp_GetStdHandle'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0xf2): undefined reference to `_imp_GetStdHandle'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x107): undefined reference to `_imp_GetStdHandle'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x133): undefined reference to `_imp_GetStartupInfoW'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x162): undefined reference to `memset'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x1ba): undefined reference to `_imp_CreateProcessW'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x1e1): undefined reference to `_imp_WaitForSingleObject'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x1f5): undefined reference to `_imp_GetExitCodeProcess'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x205): undefined reference to `_imp_CloseHandle'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x215): undefined reference to `_imp_CloseHandle'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x24e): undefined reference to `_imp_GetCommandLineW'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x270): undefined reference to `_imp_GetModuleFileNameW'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x283): undefined reference to `wcsrchr'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x296): undefined reference to `wcscpy'
osdep/win32-console-wrapper.c.25.o:win32-console-wrapper.c:(.text+0x2ab): undefined reference to `_imp_SetEnvironmentVariableW'
/usr/i686-w64-mingw32/lib/libmingw32.a(lib32_libmingw32_a-crt0_w.o): In function `wmain':
/home/qyot27/mingw-packages/mingw-w64/mingw-w64-crt/crt/crt0_w.c:23: undefined reference to `wWinMain@16'
collect2: error: ld returned 1 exit status

* Node /home/qyot27/mpv-build-deps/mpv/build/osdep/mpv-rc.o is created more than once (full message on 'waf -v -v'). The task generators are:
  1. 'objects' in /home/qyot27/mpv-build-deps/mpv
  2. 'objects' in /home/qyot27/mpv-build-deps/mpv
If you think that this is an error, set no_errcheck_out on the task instance
Waf: Leaving directory `/home/qyot27/mpv-build-deps/mpv/build'
Build failed
 -> task in 'mpv' failed with exit status 1: 
	{task 140455868516944: cprogram win32-console-wrapper.c.25.o -> mpv.com}
['/usr/bin/x86_64-w64-mingw32-gcc', '-Wl,--enable-auto-import', 'osdep/win32-console-wrapper.c.25.o', '-o/home/qyot27/mpv-build-deps/mpv/build/mpv.com', '-Wl,-Bstatic', '-Wl,-Bdynamic', '-m32', '-L/usr/i686-w64-mingw32/lib', '-municode', '-mconsole']
[mpv:$] 
```

No matter what I try (various environment variables with *git clean -xdf* between runs), `osdep/win32-console-wrapper.c` gets built in pe-x86-64 format rather than pe-i386, which causes the link error.  Normally, I would encounter the WinMain error when -m32 hadn't been set in LDFLAGS, but I was using that.  I don't have a good enough handle on the structure of wscript_build.py to really see if mpv.com and its source(s? Are there more than just `osdep/win32-console-wrapper.c`?) are somehow not getting the LDFLAGS set for it.

This is strictly an issue for 32-bit targets; the 64-bit 'native' cross environment builds everything without a hitch.  It also would be fine if the environment wasn't multilib, but the alternative (building the 32-bit cross-env in a 32-bit VM, moving it back to the 64-bit OS and merging it with the main cross-env) shouldn't be considered a reasonable option anymore.

--- a/wscript_build.py
+++ b/wscript_build.py
@@ -611,8 +611,8 @@ def build(ctx):
 
             wrapctx.env.cprogram_PATTERN = "%s.com"
             wrapflags = ['-municode', '-mconsole']
-            wrapctx.env.CFLAGS = wrapflags
-            wrapctx.env.LAST_LINKFLAGS = wrapflags
+            wrapctx.env.CFLAGS = ctx.env.CFLAGS + wrapflags
+            wrapctx.env.LAST_LINKFLAGS = ctx.env.LAST_LINKFLAGS + wrapflags
 
     if ctx.dependency_satisfied('test'):
         for test in ctx.path.ant_glob("test/*.c"):

